File: examples/25-export-html/main.go
package main

import (
	"fmt"
	"os"
	"path/filepath"

	"github.com/djinn-soul/gopptx/pkg/pptx/elements"
	"github.com/djinn-soul/gopptx/pkg/pptx/export"
	"github.com/djinn-soul/gopptx/pkg/pptx/shapes"
)

func main() {
	// 1. Create a dummy image
	imgPath := "dummy_image.png"
	createDummyImage(imgPath)
	defer os.Remove(imgPath)

	// 2. Create Presentation structure
	// Note: We use elements.SlideContent directly as per export API
	slides := []elements.SlideContent{
		elements.NewSlide("HTML Export Demo").
			AddBullet("This presentation was exported to HTML via gopptx.").
			AddBullet("It mimics the ppt-rs export style."),
		elements.NewSlide("Visual Content").
			AddShape(shapes.NewShape(shapes.ShapeTypeRectangle, 1000000, 1000000, 3000000, 1000000).
				WithText("This is a shape text.")).
			AddImage(shapes.Image{Path: imgPath, CX: 1000000, CY: 1000000}),
		elements.NewSlide("Code Simulation").
			AddShape(shapes.NewShape(shapes.ShapeTypeRectangle, 1000000, 2000000, 3000000, 2000000).
				WithText("func main() {\n    fmt.Println(\"Hello World\")\n}")),
	}

	title := "Export Demo"
	outputHTML := "examples/output/25_export.html"
	outputPDF := "examples/output/25_export.pdf"

	// Ensure output dir exists
	if err := os.MkdirAll("examples/output", 0o755); err != nil {
		fmt.Printf("Error creating output directory: %v\n", err)
		return
	}

	// 3. Export HTML
	fmt.Println("Exporting to HTML...")
	htmlContent := export.HTML(title, slides)
	if err := os.WriteFile(outputHTML, []byte(htmlContent), 0o644); err != nil {
		fmt.Printf("Error writing HTML: %v\n", err)
		return
	}
	fmt.Printf("HTML exported to %s\n", outputHTML)

	// 4. Export PDF (if LibreOffice available)
	fmt.Println("Attempting PDF export (requires LibreOffice)...")
	if err := export.PDF(title, slides, outputPDF); err != nil {
		fmt.Printf("PDF export failed (expected if LibreOffice not installed): %v\n", err)
	} else {
		fmt.Printf("PDF exported to %s\n", outputPDF)
	}
}

func createDummyImage(path string) {
	// 1x1 red pixel PNG
	data := []byte{
		0x89, 0x50, 0x4E, 0x47, 0x0D, 0x0A, 0x1A, 0x0A,
		0x00, 0x00, 0x00, 0x0D, 0x49, 0x48, 0x44, 0x52,
		0x00, 0x00, 0x00, 0x01, 0x00, 0x00, 0x00, 0x01,
		0x08, 0x06, 0x00, 0x00, 0x00, 0x1F, 0x15, 0xC4,
		0x89, 0x00, 0x00, 0x00, 0x0A, 0x49, 0x44, 0x41,
		0x54, 0x78, 0x9C, 0x63, 0xF8, 0xCF, 0x00, 0x00,
		0x02, 0x03, 0x01, 0x01, 0x24, 0x95, 0x8C, 0xFC,
		0x00, 0x00, 0x00, 0x00, 0x49, 0x45, 0x4E, 0x44,
		0xAE, 0x42, 0x60, 0x82,
	}
	_ = os.WriteFile(path, data, 0o644)
	abs, _ := filepath.Abs(path)
	fmt.Printf("Created dummy image at %s\n", abs)
}

File: examples/25-export-html/main.py
"""Demonstrate creating content for HTML export.

This example demonstrates:
- Building slides with shapes, images, and code-like text content
- Documenting the HTML export workflow
- Creating content that maps well to HTML/CSS rendering
"""

from __future__ import annotations

import tempfile
from pathlib import Path

from gopptx import Presentation
from gopptx.presentation.slides import SlideLayoutType
from gopptx.schemas import Inches

# Minimal 1x1 red pixel PNG
_RED_PNG = bytes([
    0x89,
    0x50,
    0x4E,
    0x47,
    0x0D,
    0x0A,
    0x1A,
    0x0A,
    0x00,
    0x00,
    0x00,
    0x0D,
    0x49,
    0x48,
    0x44,
    0x52,
    0x00,
    0x00,
    0x00,
    0x01,
    0x00,
    0x00,
    0x00,
    0x01,
    0x08,
    0x06,
    0x00,
    0x00,
    0x00,
    0x1F,
    0x15,
    0xC4,
    0x89,
    0x00,
    0x00,
    0x00,
    0x0A,
    0x49,
    0x44,
    0x41,
    0x54,
    0x78,
    0x9C,
    0x63,
    0xF8,
    0xCF,
    0x00,
    0x00,
    0x02,
    0x03,
    0x01,
    0x01,
    0x24,
    0x95,
    0x8C,
    0xFC,
    0x00,
    0x00,
    0x00,
    0x00,
    0x49,
    0x45,
    0x4E,
    0x44,
    0xAE,
    0x42,
    0x60,
    0x82,
])


def _add_html_export_slide(prs: Presentation) -> None:
    """Add an intro slide about HTML export."""
    prs.add_bullet_slide(
        "HTML Export Demo",
        [
            "This presentation was exported to HTML via gopptx.",
            "It mimics the ppt-rs export style.",
        ],
    )


def _add_visual_content_slide(prs: Presentation, image_path: str) -> None:
    """Add a slide with a shape and image."""
    prs.add_slide("Visual Content", layout=SlideLayoutType.BLANK)
    idx = prs.slide_count - 1

    prs.add_shape(
        idx,
        "RECTANGLE",
        bounds=(Inches(1), Inches(1), Inches(3), Inches(1)),
        text="This is a shape text.",
        properties={"fill": {"solid": "0078D4"}},
    )
    prs.add_image(idx, image_path, (Inches(5), Inches(1), Inches(1), Inches(1)))


def _add_code_simulation_slide(prs: Presentation) -> None:
    """Add a slide simulating a code block."""
    prs.add_slide("Code Simulation", layout=SlideLayoutType.BLANK)
    idx = prs.slide_count - 1

    prs.add_shape(
        idx,
        "RECTANGLE",
        bounds=(Inches(1), Inches(2), Inches(3), Inches(2)),
        text='func main() {\n    fmt.Println("Hello World")\n}',
        properties={"fill": {"solid": "F0F0F0"}},
    )


def main() -> None:
    """Create a presentation demonstrating HTML export content."""
    output_dir = Path("examples/output")
    output_dir.mkdir(exist_ok=True)

    with tempfile.NamedTemporaryFile(suffix=".png", delete=False) as tmp:
        tmp.write(_RED_PNG)
        image_path = tmp.name

    try:
        with Presentation.new("Export Demo") as prs:
            _add_html_export_slide(prs)
            _add_visual_content_slide(prs, image_path)
            _add_code_simulation_slide(prs)

            output_path = output_dir / "25-export-html.pptx"
            prs.save(str(output_path))
            print(f"Saved: {output_path}")
    finally:
        Path(image_path).unlink(missing_ok=True)

    print("\n=== SUMMARY ===")
    print("Created a 3-slide presentation for HTML export:")
    print("  - Intro slide with export description")
    print("  - Visual content with shape and image")
    print("  - Code simulation slide")


if __name__ == "__main__":
    main()
